src/Entity/Transformer.php

```diff
@@ -40,14 +47,17 @@ final class Transformer implements TransformerInterface
     private array $children = [];
 
     /**
-     * @param ManagerInterface<PageBlockInterface> $blockManager
+     * @param ManagerInterface<PageBlockInterface>                          $blockManager
+     * @param SerializerInterface&NormalizerInterface&DenormalizerInterface $serializer
      */
     public function __construct(
         private SnapshotManagerInterface $snapshotManager,
         private PageManagerInterface $pageManager,
         private ManagerInterface $blockManager,
-        private ManagerRegistry $registry
+        private ManagerRegistry $registry,
+        private $serializer
```

Comment:
its already in the Branch for upcoming (switched from 4.x to 5.x)

Reply:
It's not because it's 5.x that we should introduce HARD BC break when it could be SOFT BC break.
We could add an optional argument to the constructor in 4.x.

But looking at the PR I'm start wondering what were the reason for not doing this on 4.x ?
Do you see any blocking point for introducing this without BC break in 4.x ?

You could do
```
public function __construct(
        private SnapshotManagerInterface $snapshotManager,
        private PageManagerInterface $pageManager,
        private ManagerInterface $blockManager,
        private ManagerRegistry $registry
        private ?Serializer $serializer = null
    ) {
    }
````

And in the code
```
if ($this->serializert) {
     // use it
} else {
     // old way
}
```

Reply:
i will try to redo the old way after i fixed the psalm and phpstan bugs with "numeric-string", it seems it doesn't like that?

or is that a psalm bug for the Datetime->format return type?

Reply:
It's right that format can return any string. So we cannot be sure it's numeric-string.

But a return-type provider could be implemented in psalm
```
$datetime->format('U')
```
is always a numeric-string.

Adding the phpdoc like you did seems an ok way so far.

Reply:
it seems the numeric string check wasn't needed anyway so it seems psalm already had something like this?

Reply:
nice

Reply:
it seems it didn't work, i might do an issue for psalm because i don't like the extra `@var numeric-string` in the code :/

Reply:
found the reason of my confusion ... phpdoc aka PHPStan understands that format("U") returns numeric-string
but psalm doesn't yet, that's why my IDE got confused on if the parameter are needed or not